adobe/manip.hpp

```diff
@@ -18,11 +18,17 @@
 
 #include <iostream>
 
+/*
+    REVISIT (sean-parent): As far as I can determine, this file does not use adobe/serializable.hpp
+    However, if it is not included here, the build fails for C++17 under MSVC only. I have not been
+    able to determine why this is the case.
+*/
+#include <adobe/serializable.hpp>
```

Comment:
@dabrahams - I was able to significantly reduce the problem to the pointer where I determined that `adobe/serializable.hpp` had to be included before `manip.hpp`, but there does not appear to be any dependency on `serializable.hpp` - circular or not. In fact, in my reduced case, not include `manip.hpp` at all also worked. If it was more than one configuration of one compiler, I'd reduce it more.

Wihtout this line the following fails to compiler:
```
#include <adobe/array.hpp>
#include <adobe/manip.hpp>
#include <adobe/serializable.hpp>

/**************************************************************************************************/

namespace adobe {

/**************************************************************************************************/

void test() {
    serializable_t value{array_t{}};
}

/**************************************************************************************************/

} // namespace adobe
```

Reply:
Without what line?

I don't think giving up on the reduction is appropriate.  This configuration of this compiler applies all Adobe products and many others.  IMO you should get to the point where you can submit a bug report.  But that's just me.

Reply:
Found it - a namespace lookup issue. Fixed property in array.hpp.